R/covariancecorrelation.R

```diff
@@ -1,23 +1,58 @@
 # Compute a matrix containing the pairwise weighted correlations between each column in
-# 'data'.
-weightedPartialCovarianceMatrix <- function(data, weights, correlation = FALSE)
+# 'X'.
+weightedPartialCovarianceMatrix <- function(X, weights, correlation = FALSE)
 {
-    num.cols <- ncol(data)
+    # Removing missing or inappropriate weights.
+    valid <- !is.na(weights) & weights > 0
+    weights <- weights[valid]
+    X <- X[valid, ]
+    # Remove entirely missing rows
+    if (any(all.missing <- apply(X, 1, function(x) all(is.na(x)))))
+    {
+        not.all.missing <- !all.missing
+        X <- X[not.all.missing, ]
+        weights <- weights[not.all.missing]
+    }
+    # Use fast algorithms if no remaining missing values in X
+    if (!anyNA(X))
+    {
+        if (all(weights[-1] == weights[1]))
```

Comment:
Subsetting with `weights[-1]` is surely slower than just doing the extra comparison with no subsetting. This would also break if weights has length 1.

Reply:
yep amended now. Not sure the weights of length 1 would ever arise since weights is typically always on length `n`. However, the code doesnt check this anywhere from what I have seen. Fixed in https://github.com/Displayr/flipStatistics/commit/f2b3e132965add46e9c25543f4195abb8832c480 regardless.